layer/runs/initializer.py

```diff
@@ -0,0 +1,121 @@
+import atexit
+import logging
+import os
+import uuid
+from pathlib import Path
+from typing import Optional, Sequence, Set
+
+from layer.clients.layer import LayerClient
+from layer.contracts.fabrics import Fabric
+from layer.contracts.project_full_name import ProjectFullName
+from layer.contracts.projects import Project, ProjectLoader
+from layer.contracts.runs import Run, RunStatus
+from layer.runs import context
+
+
+LAYER_RUN_ID_ENV_VARIABLE = "LAYER_RUN_ID"
+
+logger = logging.Logger(__name__)
+
+
+class RunInitializer:
+    """
+    Class responsible for setting up a run
+    and creating a project in the Layer Backend if it doesn't exist already
+    """
+
+    def __init__(self, layer_client: LayerClient):
+        self._layer_client = layer_client
+
+    def setup_project(
+        self,
+        project_full_name: ProjectFullName,
+        run_name: Optional[str] = None,
+        run_index: Optional[int] = None,
```

Comment:
i thought `index` was auto-incremented by the BE? why is it an argument?

Reply:
it's an argument in case a user wants to resume an existing run. this was indeed your suggestion the other day 😄